### PR TITLE
AWS auto case - byo IAM profile

### PR DESCRIPTION
**Pull request by yunjiang29**
OCP-75511	[ipi-on-aws] Create cluster by using existing IAM profiles (master only)
OCP-75512	[ipi-on-aws] Create cluster by using existing IAM profiles (worker only)
OCP-75514	[ipi-on-aws] Create cluster by using existing IAM profiles (defaultMachinePlatform)

* New jobs
`periodic-ci-openshift-verification-tests-master-installation-nightly-4.18-aws-ipi-byo-iam-profile-default-f28`
`periodic-ci-openshift-verification-tests-master-installation-nightly-4.18-aws-ipi-byo-iam-profile-master-f28`
`periodic-ci-openshift-verification-tests-master-installation-nightly-4.18-aws-ipi-byo-iam-profile-worker-f28`

* Enhance post-check `cucushift-installer-check-aws-byo-iam-profile`

### CODE REVIEW

**Comment by yunjiang29** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installation-nightly-4.18-aws-ipi-byo-iam-profile-default-f28

**Comment by yunjiang29** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installation-nightly-4.18-aws-ipi-byo-iam-profile-master-f28

**Comment by yunjiang29** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installation-nightly-4.18-aws-ipi-byo-iam-profile-worker-f28

**Comment by yunjiang29** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installation-nightly-4.18-aws-ipi-byo-iam-profile-master-f28

**Comment by yunjiang29** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installation-nightly-4.18-aws-ipi-byo-iam-profile-worker-f28

**Comment by yunjiang29** [comment]
/pj-rehearse pull-ci-openshift-installer-release-4.19-e2e-aws-ovn-custom-iam-profile

**Comment by yunjiang29** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installation-nightly-4.18-aws-ipi-byo-iam-profile-default-f28

**Comment by yunjiang29** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installation-nightly-4.18-aws-ipi-byo-iam-profile-master-f28

**Comment by yunjiang29** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installation-nightly-4.18-aws-ipi-byo-iam-profile-worker-f28

**Comment by yunjiang29** [comment]
/pj-rehearse pull-ci-openshift-installer-release-4.19-e2e-aws-ovn-custom-iam-profile

**Comment by yunjiang29** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.18-multi-nightly-aws-ipi-byo-iam-profile-arm-f7

**Comment by yunjiang29** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installation-nightly-4.18-aws-ipi-byo-iam-profile-default-f28

**Comment by yunjiang29** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installation-nightly-4.18-aws-ipi-byo-iam-profile-master-f28

**Comment by yunjiang29** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installation-nightly-4.18-aws-ipi-byo-iam-profile-worker-f28

**Comment by yunjiang29** [comment]
/pj-rehearse pull-ci-openshift-installer-release-4.19-e2e-aws-ovn-custom-iam-profile

**Comment by yunjiang29** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.18-multi-nightly-aws-ipi-byo-iam-profile-arm-f7

**Comment by yunjiang29** [comment]
/pj-rehearse pull-ci-openshift-installer-release-4.18-e2e-aws-ovn-custom-iam-profile

**Comment by yunjiang29** [comment]
/pj-rehearse pull-ci-openshift-installer-release-4.17-e2e-aws-ovn-custom-iam-profile

**Comment by yunjiang29** [comment]
Tests:
* [PASS] https://qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/view/gs/qe-private-deck/pr-logs/pull/openshift_release/58256/rehearse-58256-periodic-ci-openshift-verification-tests-master-installation-nightly-4.18-aws-ipi-byo-iam-profile-default-f28/1851496446763732992
* [PASS] https://qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/view/gs/qe-private-deck/pr-logs/pull/openshift_release/58256/rehearse-58256-periodic-ci-openshift-verification-tests-master-installation-nightly-4.18-aws-ipi-byo-iam-profile-master-f28/1851496404426428416
* [PASS] https://qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/view/gs/qe-private-deck/pr-logs/pull/openshift_release/58256/rehearse-58256-periodic-ci-openshift-verification-tests-master-installation-nightly-4.18-aws-ipi-byo-iam-profile-worker-f28/1851496495790952448
* [PASS] https://qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/view/gs/qe-private-deck/pr-logs/pull/openshift_release/58256/rehearse-58256-periodic-ci-openshift-openshift-tests-private-release-4.18-multi-nightly-aws-ipi-byo-iam-profile-arm-f7/1851496502090797056

**Comment by yunjiang29** [comment]
@jianlinliu PTAL

File: ci-operator/config/openshift/verification-tests/openshift-verification-tests-master__installation-nightly-4.18.yaml [hunk 1]
```diff
@@ -56,6 +56,51 @@ resources:
       cpu: 400m
       memory: 800Mi
 tests:
+- as: aws-ipi-byo-iam-profile-master-f28
```
Comment (jianlinliu) [inline ci-operator/config/openshift/verification-tests/openshift-verification-tests-master__installation-nightly-4.18.yaml:59]: I think maybe we can do some similar things like `azure-ipi-confidential-trustedlaunch` in regular prow ci jobs. 

Enable byo IAM profile on cp and compute in e2e job, and Enable byo IAM profile on default machine platform in destructive e2e job. 

If you want to cover byo-iam-on-only-cp (case-1), byo-iam-on-only-compute (case-2), byo-iam-on-default-machine (case-3), then we can re-use the existing regular ci jobs to cover case-1 and case-2, here we only need to add one more job to cover case-3. WDYT?

File: ci-operator/step-registry/ipi/conf/aws/byo-iam-profile/ipi-conf-aws-byo-iam-profile-ref.yaml [hunk 2]
```diff
@@ -10,5 +10,15 @@ ref:
     requests:
       cpu: 10m
       memory: 100Mi
+  env:
+  - name: ENABLE_BYO_IAM_PROFILE_ON_CUMPUTE_NODE
```
Comment (jianlinliu) [inline ci-operator/step-registry/ipi/conf/aws/byo-iam-profile/ipi-conf-aws-byo-iam-profile-ref.yaml:14]:
```suggestion
  - name: ENABLE_BYO_IAM_PROFILE_CUMPUTE
```

For shorter var name

File: ci-operator/step-registry/ipi/conf/aws/byo-iam-profile/ipi-conf-aws-byo-iam-profile-ref.yaml [hunk 3]
```diff
@@ -10,5 +10,15 @@ ref:
...
+    default: "yes"
+  - name: ENABLE_BYO_IAM_PROFILE_ON_CONTROL_PLANE_NODE
```
Comment (jianlinliu) [inline ci-operator/step-registry/ipi/conf/aws/byo-iam-profile/ipi-conf-aws-byo-iam-profile-ref.yaml:16]:
```suggestion
  - name: ENABLE_BYO_IAM_PROFILE_CONTROL_PLANE
```
For shorter var name

File: ci-operator/step-registry/ipi/conf/aws/byo-iam-profile/ipi-conf-aws-byo-iam-profile-ref.yaml [hunk 4]
```diff
@@ -10,5 +10,15 @@ ref:
...
+    default: "yes"
+  - name: ENABLE_BYO_IAM_PROFILE_ON_DEFAULT_MACHINE_PLATFORM
```
Comment (jianlinliu) [inline ci-operator/step-registry/ipi/conf/aws/byo-iam-profile/ipi-conf-aws-byo-iam-profile-ref.yaml:18]:
```suggestion
  - name: ENABLE_BYO_IAM_PROFILE_DEFAULT_MACHINE
```
For shorter var name

File: ci-operator/config/openshift/verification-tests/openshift-verification-tests-master__installation-nightly-4.18.yaml (hunk 1 above)
Comment (yunjiang29) [reply ci-operator/config/openshift/verification-tests/openshift-verification-tests-master__installation-nightly-4.18.yaml:59]: e2e `cp+compute`, destructive e2e `default`
installation jobs: `cp-only` and `compute-only`
```
- as: aws-ipi-byo-iam-profile-arm-f7
  cron: 6 11 6,15,22,29 * *
  steps:
    cluster_profile: aws-qe
    env:
      BASE_DOMAIN: qe.devcluster.openshift.com
      COMPUTE_NODE_TYPE: m6g.xlarge
      CONTROL_PLANE_INSTANCE_TYPE: m6g.xlarge
      OCP_ARCH: arm64
    test:
    - chain: openshift-e2e-test-qe
    workflow: cucushift-installer-rehearse-aws-ipi-byo-iam-profile
- as: aws-ipi-byo-iam-profile-fips-amd-f28-destructive
  cron: 55 14 22 * *
  steps:
    cluster_profile: aws-qe
    env:
      BASE_DOMAIN: qe.devcluster.openshift.com
      ENABLE_BYO_IAM_PROFILE_CONTROL_PLANE: "false"
      ENABLE_BYO_IAM_PROFILE_CUMPUTE: "false"
      ENABLE_BYO_IAM_PROFILE_DEFAULT_MACHINE: "true"
      FIPS_ENABLED: "true"
    test:
    - chain: openshift-e2e-test-qe-destructive
    workflow: cucushift-installer-rehearse-aws-ipi-byo-iam-profile
```

**Comment by jianlinliu** [comment]
cc @liangxia to review

File: ci-operator/config/openshift/verification-tests/openshift-verification-tests-master__installation-nightly-4.18.yaml (hunk 1 above)
Comment (jianlinliu) [reply ci-operator/config/openshift/verification-tests/openshift-verification-tests-master__installation-nightly-4.18.yaml:59]: looks good to me.

@yunjiang29 shall we cover them in 4.17 ?

Comment (yunjiang29) [reply ci-operator/config/openshift/verification-tests/openshift-verification-tests-master__installation-nightly-4.18.yaml:59]: 417 https://github.com/openshift/release/pull/58383

**Pull request merged**